### PR TITLE
fix: add deprecated methods to History type

### LINKED ISSUE #7: TS : history definition error
Hi,

I use React with TS (I use [react-scripts-ts](https://github.com/wmonk/create-react-app-typescript) and when I define the Router history, I've got this error :
```
Error in ./src/index.tsx
(32,15): error TS2322: Type 'SynchronizedHistory' is not assignable to type 'History | undefined'.
  Type 'SynchronizedHistory' is not assignable to type 'History'.
    Types of property 'listenBefore' are incompatible.
      Type '(hook: TransitionHook) => UnsubscribeCallback' is not assignable to type '(hook: TransitionHook) => () => void'.
        Types of parameters 'hook' and 'hook' are incompatible.
          Type 'History.History.TransitionHook' is not assignable to type 'MobxReactRouter.TransitionHook'.
            Types of parameters 'location' and 'location' are incompatible.
              Type 'MobxReactRouter.Location' is not assignable to type 'History.Location'.
                Property 'query' is optional in type 'Location' but required in type 'Location'.
```

It comes from this line : `<Router history={history} >`
I did exactly like the documentation : `const history = syncHistoryWithStore(browserHistory, routingStore);
`

Thank you for your work !

### PR DESCRIPTION
**Pull request by rokoroku**
it fixes #7

### CODE REVIEW

File: types.d.ts [hunk 1]
```diff
@@ -34,16 +34,33 @@ declare namespace MobxReactRouter {
...
   export type Location = {
     pathname: string;
     search: string;
     action: string;
     key: string;
-    hash: string
     state: any;
-    query?: any;
+    query: any;
+    hash?: string
```
Comment (alisd23) [inline types.d.ts]: Just realised there's a semi-colon missing here

Comment (rokoroku) [inline types.d.ts]: Oops, you're right. 👍

**Comment by rokoroku** [comment]
Added ``@types/react-router`` to the devDependencies for testing with ``react-router``.
(v2.0.41 is currently the latest type definition for react-router. I will keep watching for updates.)

**Comment by alisd23** [comment]
Looks good, thanks again 😀

**Pull request merged**